bin/compare-factory-config.py

```diff
@@ -176,27 +192,33 @@ def run(argv):
     gfactory = []
     gfactory.extend(glob.glob(os.path.abspath(temp_dir) + '/*.xml')
                     + (glob.glob(os.path.abspath(temp_dir) + '/OSG_autoconf/*.yml')))
-    # dictionary that stores (GLIDEIN_ResourceNames: entry name) pairs
+    # dictionary that stores (GLIDEIN_ResourceNames: (entry name, suggestion)) pairs
     gfactory_DB = {}
     if factory_dump:
         print(f'\nAll the GLIDEIN_ResourceNames in factory: \n')
     for xml in gfactory:
         get_gfactory_data(gfactory_DB, xml)
 
+    
+    
     # finding results
+    # extract resources from tuples
+    resources = set([x[0] for x in topology_DB['resources']])
     # compairing gfactory with Topology resources
     match_nonresource_entries = find_non_resource_matches(
-        gfactory_DB, topology_DB)
+        gfactory_DB, topology_DB, resources)
     # Entry names corresponding to GLIDEIN_ResourceNames above
-    nonmatch_all_entries = find_non_topology_matches(gfactory_DB, topology_DB)
+    nonmatch_all_entries = find_non_topology_matches(
+        gfactory_DB, topology_DB, resources)
 
     # output formatted results
+    print(f'\nOutput format: <GLIDEIN_ResourceName>:<corresponding factory entry name>:<suggestion for a resource match>\n')
```

Comment:
Is this the actual format? If so, could you switch the first and second columns?

Reply:
that's a mistake here, output was correct and description is wrong